finality-provider/config/config.go

```diff
@@ -26,15 +26,35 @@ const (
 	DefaultRPCPort                     = 12581
 	defaultConfigFileName              = "fpd.conf"
 	defaultNumPubRand                  = 50000 // support running of roughly 5 days with block production time as 10s
-	defaultNumPubRandMax               = 500000
-	defaultTimestampingDelayBlocks     = 6000 // 100 BTC blocks * 600s / 10s
-	defaultBatchSubmissionSize         = 1000
-	defaultRandomInterval              = 30 * time.Second
+	defaultBatchSubmissionSize         = 100
 	defaultSubmitRetryInterval         = 1 * time.Second
 	defaultSignatureSubmissionInterval = 1 * time.Second
 	defaultMaxSubmissionRetries        = 20
 	defaultBitcoinNetwork              = "signet"
 	defaultDataDirname                 = "data"
+	// TimestampingDelayBlocks is the delay for a commit to become available due to BTC timestamping protocol
+	// it is calculated by converting BTC block time (parameter w=300) to Babylon Genesis block time
+	// 300 BTC blocks * 600s / 10s where 300 BTC blocks is the system parameter of BTC block time required
+	// by the btc timestamping protocol.
+	// for testnet with w=100, the recommended default value is 6000
+	defaultTimestampingDelayBlocks = 18000
+)
+
+// Constants for system parameters validation limits
+const (
+	// MaxBatchSize is the maximum allowed batch submission size,
+	// this limits BatchSubmissionSize as larger values could cause error due to insufficient fees
+	MaxBatchSize = 100
+	// MaxPubRand is the maximum allowed number of public randomness in each commit,
+	// this limits NumPubRand as larger values could take more than 10min to generate/store
+	// rand proofs
+	MaxPubRand = 500000 //
+	// MinPubRand is the minimum allowed number of public randomness in each commit
+	// a commit with less than this value will be rejected by Babylon Genesis
+	MinPubRand = 8192
```

Comment:
I think we should increase this. `8192` lasts only for ~22 hours, which would be below the required threshold for mainnet. We could set it to something like 18k or 16384

Reply:
lets set it to 18k  👍

Reply:
after second thought, maybe `MinPubRand` should be somehow based on `TimestampingDelayBlocks` 🤔 It seems the larger `TimestampingDelayBlocks` the larger `MinPubRand` should be.

Reply:
I set it to 18k and merging this PR.

> after second thought, maybe MinPubRand should be somehow based on TimestampingDelayBlocks 🤔 It seems the larger TimestampingDelayBlocks the larger MinPubRand should be.

I don't see a clear relation like this. I think as long as `NumPubRand` is large enough to cover the inaccuracy (btc being slow and randomness might run out when the commit is timestamped) of btc time estimation is OK.